tangelo/linq/translator/translate_openqasm.py

```diff
@@ -57,12 +60,47 @@ def translate_openqasm(source_circuit):
     Returns:
         str: the corresponding OpenQASM program, as per IBM Qiskit.
     """
-    from .translate_qiskit import translate_qiskit
+    return translate_c_to_openqasm(source_circuit)
+
+
+@deprecated("Please use the translate_circuit function.")
+def _translate_openqasm2abs(source_circuit):
+    """Take an OpenQASM 2.0 string as input (as defined by IBM Qiskit), return
+    the equivalent abstract circuit. Only a subset of OpenQASM supported, mostly
+    to be able to go back and forth QASM and abstract representations to
+    leverage tools and innovation implemented to work in the QASM format. Not
+    designed to support elaborate QASM programs defining their own operations.
+    Compatible with qiskit.QuantumCircuit.from_qasm method.
+
+    Assumes single-qubit measurement instructions only. Final qubit register
+    measurement is implicit.
+
+    Args:
+        openqasm_string(str): an OpenQASM program, as a string, as defined by
+            IBM Qiskit.
+
+    Returns:
+        Circuit: corresponding quantum circuit in the abstract format.
+    """
+    return translate_c_from_openqasm(source_circuit)
+
+
+def translate_c_to_openqasm(source_circuit):
+    """Take in an abstract circuit, return a OpenQASM 2.0 string using IBM
+    Qiskit (they are the reference for OpenQASM).
+
+    Args:
+        source_circuit: quantum circuit in the abstract format.
+
+    Returns:
+        str: the corresponding OpenQASM program, as per IBM Qiskit.
+    """
+    from .translate_qiskit import translate_c_to_qiskit
 
-    return translate_qiskit(source_circuit).qasm()
+    return translate_c_to_qiskit(source_circuit).qasm()
 
 
-def _translate_openqasm2abs(openqasm_str):
+def translate_c_from_openqasm(openqasm_str):
```

Comment:
Open question: this explicit conversion only supported a subset of OpenQASM, and was written early 2020.

Can we simply implement this using the reverse qiskit translation combined with https://qiskit.org/documentation/stubs/qiskit.circuit.QuantumCircuit.from_qasm_str.html#qiskit.circuit.QuantumCircuit.from_qasm_str ?

If this is tricky, you can leave this alone for this PR. I suspect some of our circuit data for tests was generated in a particular quasi format, which means things may break. In that case, a follow-up PR can do that.